modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Genesis.scala

```diff
@@ -96,24 +102,44 @@ object Genesis {
       _ <- logger.info(s"Genesis binary ${binaryHash.show} and ${incrementalBinaryHash.show} accepted and sent to Global L0")
     } yield ()
 
-    def accept(
-      path: Path,
-      dataApplication: Option[BaseDataApplicationL0Service[F]] = None
-    )(implicit context: L0NodeContext[F]): F[Unit] = {
+    override def accept(dataApplication: Option[BaseDataApplicationL0Service[F]])(genesisPath: Path)(
+      implicit context: L0NodeContext[F]
+    ): F[Unit] = genesisLoader
+      .loadSignedGenesis(genesisPath)
+      .flatTap { genesis =>
+        dataApplication
+          .map(app => app.setCalculatedState(genesis.ordinal, app.genesis.calculated))
+          .getOrElse(false.pure[F])
```

Comment:
`flatTap` expects `F[_]` to be returned but `dataApplication.map(...)` is `Option[F[_]]` so we need to get rid of `Option` to make the compiler happy. I think we can just use `traverse` instead of `getOrElse`.

```
      _ <- dataApplication.traverse { app =>
            app.setCalculatedState(firstIncrementalSnapshot.ordinal, app.genesis.calculated)
          }
```